store/pogreb/pogreb.go

```diff
@@ -161,37 +249,58 @@ func (s *pStorage) RemoveProvider(providerID peer.ID) error {
 	panic("not implemented")
 }
 
-// DuplicateEntry checks if the entry already exists in the index. An entry
-// for the same provider but different metadata is not considered
-// a duplicate entry,
-func duplicateEntry(in entry.Value, old []entry.Value) bool {
-	for i := range old {
-		if in.Equal(old[i]) {
-			return true
-		}
+func (s *pStorage) decreaseRefC(k []byte) (bool, error) {
+	old, found, err := s.getEntry(k)
```

Comment:
Race condition?  If concurrent calls to `decreaseRef` get the same value for `old,.RefC` and they all decrease it to the same value, instead of by one for each caller.